common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java

```diff
@@ -99,9 +106,9 @@ public BrokerRequest brokerRequestFromAcquireTokenParameters(@NonNull final Acqu
                         parameters.getExtraQueryStringParameters() != null ?
                                 QueryParamsAdapter._toJson(parameters.getExtraQueryStringParameters())
                                 : null
-                ).prompt(parameters.getOpenIdConnectPromptParameter().name())
+                ).prompt(parameters.getPrompt().name())
                 .claims(parameters.getClaimsRequestJson())
-                .forceRefresh(parameters.getForceRefresh())
+                .forceRefresh(false)
```

Comment:
This would break `claimsRequest` flow. Before this change , `forceRefresh` was on `OperationParameter` [here ](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/request/OperationParameters.java#L85) and we set `forceRefresh` to true for interactive calls when `claimsRequest` is present [here ](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java#L147). This is done because, in broker we would attempt to silently get a token for interactive calls as well if we a matching account using PRT, FRT, and if `claimsRequest` is present we would need to skip cache and refresh the token.

So i think you should move back `forceRefresh` to base class and add the logic back

Reply:
@kreedula `forceRefresh` was indeed set on base class operation parameters but I believe it was never used in any interactive flow and hence I removed it and put it only on silent. Can you share the code snippet where this flow is actually being used during an interactive call?

Reply:
Okay, I see so it seems it was never used on the `OperationParameters` but we need it on the `BrokerRequest` object to decide if we need to skip cache or not.

Reply:
@kreedula I don't think I need to put them on base command parameter class in that case. I think I can just do [this ](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/main/java/com/microsoft/identity/client/internal/controllers/OperationParametersAdapter.java#L147)logic here at the adapter level and set the value of `forceRefresh` on `BrokerRequest` based on that.

Reply:
It should work with the current implementation, as we unpack the BrokerRequest twice , once [here  ](https://github.com/AzureAD/ad-accounts-for-android/blob/dev/AADAuthenticator/src/msal/java/com/microsoft/identity/client/BrokerClientApplication.java#L349) and the other one to convert to `BrokerAcquireTokenParameters`. I would like to make some changes to avoid unpacking `BrokerRequest` twice especially with compressed payloads, have a work item for it. So it should be on `TokenCommandParameters` so that when we transform `BrokerRequest` to `BrokerAcquireTokenCommandParameters` , we would have the value to set and get

Reply:
@kreedula That doesn't make sense to me. All you need is for `forceRefresh` to be available on `BrokerRequest` regardless of how many times we unpack it. Why do you need it on `TokenCommandParameters`? When we skip cache in broker, then we create silent token command parameters and not interactive so you shouldn't need it to be on either interactive parameters or the base class. Would there be a case where we would need `forceRefresh` to live on interactive parameters? Am i missing something here?

Reply:
let's chat offline, so we can resolve this faster

Reply:
Resolved offline, consensus is to `forceRefresh` to `BrokerRequest` based on `claimsRequest`  and leave it out of `TokenCommandParameters`

Reply:
Per second discussion decided to go with @kreedula suggestion to put `force refresh` on base token command parameters as we would need to send it to broker request based on the value of Claims Request passed in by the developer and not the merged claims request as that could contain client capabilities.

Addressed it in common here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/880/commits/d5674111b5f49ff0701ffdc437b0baed90848cb5
MSAL here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/993/commits/69984b2d5e9f76c4b686d4e0da42b60bf03186f0
Added back tests here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/993/commits/13a7dae3b6490d88cfc3d6b22917dcb7100f5a09